project/BensonYang1999/readme.md

```diff
@@ -0,0 +1,57 @@
+# Simple Machine Learning Toolkit
+
+## Basic Information
+A simple library for machine learning, such as regression, classification, clustering.
+
+## Problem to Solve
+I'm trying to build a library in C++ to make machine learning faster and multipurpose.
+
+There are lots of tools for machine learning in python, but they may not be as efficient as using c++. So I'm going to build a machine laarning library in c++ to make it run faster and easily used for C++ users.
+
+
+## Prospective Users
+Anyone who want to use machine learning in either C++ or Python to solve various problems.
+
+
+## System Architecture
+Use `C++` to implement various algorithm
+
+Users can choice using `Python` or `C++` to read data
+
+Use `pybind11` to convert `C++ library` to `Python library` 
+
+```mermaid
+graph TD
+    A[Data] --> B[C++ reading]
+    A --> F[Python reading]
+    B --> C[C++ preprocessing]
+    F --> G[Python preprocessing]
+    C --> D[C++ Machine Learning algorithm]
+    G --> D
+    D --> E[C++ output]
+    D --> H[Python output]
+
+```
+
+## Engineering Infrastructure
+ - Build System: make
+ - Version Control: git
+ - Test Framework: pytest
+
+
+## Schedule
+ - Week 0 (11/02): project planning and writing proposal
+ - Week 1 (11/07): construct c++ class structure
+ - Week 2 (11/14): add regression implementation
+ - Week 3 (11/21): add classification implementation
+ - Week 4 (11/28): add clustering implementation
+ - Week 5 (12/05): make some test case for testing
+ - Week 6 (12/12): test and optimize every algorithm
```

Comment:
Testing at the last stage will give you little time to fix issues found.

Reply:
I will test every approach in each week.

Reply:
Sounds good.